pandas/tests/series/test_dtypes.py

```diff
@@ -506,6 +506,33 @@ def test_is_homogeneous_type(self):
         assert Series([1, 2])._is_homogeneous_type
         assert Series(pd.Categorical([1, 2]))._is_homogeneous_type
 
+    @pytest.mark.parametrize(
+        "data, uniques, dtype",
+        [
+            ([1, 2, 2], [1, 2], "int8"),
+            ([1, 2, 2], [1, 2], "int16"),
+            ([1, 2, 2], [1, 2], "int32"),
+            ([1, 2, 2], [1, 2], "int64"),
+            ([1, 2, 2], [1, 2], "uint8"),
+            ([1, 2, 2], [1, 2], "uint16"),
+            ([1, 2, 2], [1, 2], "uint32"),
+            ([1, 2, 2], [1, 2], "uint64"),
+            ([1, 2, 2], [1.0, 2.0], "float16"),
+            ([1, 2, 2], [1.0, 2.0], "float32"),
+            ([1, 2, 2], [1.0, 2.0], "float64"),
+            ([1, 2, 2], [1.0, 2.0], "complex64"),
+            ([1, 2, 2], [1.0, 2.0], "complex128"),
+            ([True, True, False], [True, False], "bool"),
```

Comment:
I would really try to use the existing fixtures for this, e.g. any_numpy_dtype (and just cast the input in numpy before construction), skipping those which don't make sense.

Reply:
Thanks, @jreback - now using `np.sctypes` to extract all dtypes and skipping `[bytes, str, np.void]`.